sql/logictest: TestSqlLiteLogic failed

[(sql/logictest).TestSqlLiteLogic failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1912272&tab=buildLog) on [master@61f18db7dd9a054d9a4648f67546202f760b5000](https://github.com/cockroachdb/cockroach/commits/61f18db7dd9a054d9a4648f67546202f760b5000):

```
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/view/10/slt_good_4.test: 5716 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/in/100/slt_good_2.test: 6193 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/view/10/slt_good_2.test: 5693 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/view/10/slt_good_3.test: 5391 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/between/1/slt_good_0.test: 8118 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/view/10/slt_good_1.test: 5341 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/view/10/slt_good_4.test: 5755 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/in/100/slt_good_2.test: 6289 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/view/100/slt_good_4.test: 5342 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/view/10/slt_good_2.test: 5735 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/view/10/slt_good_3.test: 5430 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/between/1/slt_good_0.test: 8160 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/view/10/slt_good_1.test: 5378 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/view/10/slt_good_4.test: 5793 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/in/100/slt_good_2.test: 6327 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/view/100/slt_good_4.test: 5381 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/view/10/slt_good_2.test: 5759 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/view/10/slt_good_3.test: 5469 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/between/1/slt_good_0.test: 8214 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/view/10/slt_good_1.test: 5407 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/view/10/slt_good_4.test: 5819 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/in/100/slt_good_2.test: 6340 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/view/100/slt_good_4.test: 5420 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/view/10/slt_good_2.test: 5792 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/view/10/slt_good_3.test: 5506 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/between/1/slt_good_0.test: 8238 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/view/10/slt_good_1.test: 5447 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/in/100/slt_good_2.test: 6364 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/view/10/slt_good_4.test: 5854 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/view/100/slt_good_4.test: 5457 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/view/10/slt_good_2.test: 5827 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/view/10/slt_good_3.test: 5544 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/between/1/slt_good_0.test: 8296 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/view/10/slt_good_1.test: 5480 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/in/100/slt_good_2.test: 6400 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/view/100/slt_good_4.test: 5481 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/view/10/slt_good_4.test: 5886 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/select4.test: 2965 statements/queries
--- done: /go/src/github.com/cockroachdb/sqllogictest/test/select4.test with config fakedist-disk: 2965 tests, 1 failures
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/view/10/slt_good_2.test: 5864 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/view/10/slt_good_3.test: 5573 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/between/1/slt_good_0.test: 8360 statements/queries
        --- FAIL: TestSqlLiteLogic/fakedist-disk/select4.test (241.82s)
            logic.go:1879: 
                 pq: this query requires additional disk space: in-mem temp storage: disk budget exceeded: 1048576 bytes requested, 104857600 currently allocated, 104857600 bytes in budget
            logic.go:2458: 
                /go/src/github.com/cockroachdb/sqllogictest/test/select4.test:38818: error while processing
            logic.go:2458: /go/src/github.com/cockroachdb/sqllogictest/test/select4.test:38818: too many errors encountered, skipping the rest of the input
=== RUN   TestSqlLiteLogic/fakedist-disk
    --- FAIL: TestSqlLiteLogic/fakedist-disk (0.01s)

```

<details><summary>More</summary><p>


```
make stressrace TESTS=TestSqlLiteLogic PKG=./pkg/sql/logictest TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestSqlLiteLogic.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(sql/logictest).TestSqlLiteLogic failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1914602&tab=buildLog) on [master@a481ee62aba2ca146c3c90155f393beaa591ddda](https://github.com/cockroachdb/cockroach/commits/a481ee62aba2ca146c3c90155f393beaa591ddda):

```
rewrote:
CREATE TABLE t9 (a9 INT8, b9 INT8, c9 INT8, d9 INT8, e9 INT8, x9 VARCHAR(30), FAMILY (c9), FAMILY (e9, d9), FAMILY (b9, x9), FAMILY (a9));


--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/orderby_nosort/10/slt_good_33.test: 8356 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/orderby_nosort/10/slt_good_29.test: 6610 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/orderby_nosort/10/slt_good_3.test: 5705 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/orderby_nosort/10/slt_good_31.test: 8144 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/orderby_nosort/10/slt_good_34.test: 9443 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/view/10/slt_good_4.test: 377 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/orderby_nosort/10/slt_good_30.test: 9285 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/select4.test: 698 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/orderby_nosort/10/slt_good_3.test: 5812 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/orderby_nosort/10/slt_good_33.test: 8582 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/orderby_nosort/10/slt_good_29.test: 6735 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/orderby_nosort/10/slt_good_34.test: 9545 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/orderby_nosort/10/slt_good_31.test: 8154 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/view/10/slt_good_4.test: 449 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/orderby_nosort/10/slt_good_30.test: 9313 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/select4.test: 1021 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/orderby_nosort/10/slt_good_3.test: 5926 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/orderby_nosort/10/slt_good_33.test: 8649 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/orderby_nosort/10/slt_good_34.test: 9706 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/orderby_nosort/10/slt_good_29.test: 6785 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/view/10/slt_good_4.test: 513 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/orderby_nosort/10/slt_good_31.test: 8163 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/orderby_nosort/10/slt_good_30.test: 9363 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/orderby_nosort/10/slt_good_3.test: 6030 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/orderby_nosort/10/slt_good_34.test: 9798 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/orderby_nosort/10/slt_good_33.test: 8719 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/select4.test: 1029 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/view/10/slt_good_4.test: 586 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/orderby_nosort/10/slt_good_29.test: 6886 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/orderby_nosort/10/slt_good_31.test: 8173 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/orderby_nosort/10/slt_good_30.test: 9407 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/orderby_nosort/10/slt_good_3.test: 6162 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/orderby_nosort/10/slt_good_34.test: 9985 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/orderby_nosort/10/slt_good_33.test: 8866 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/select4.test: 1046 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/view/10/slt_good_4.test: 655 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/orderby_nosort/10/slt_good_29.test: 6926 statements/queries
--- done: /go/src/github.com/cockroachdb/sqllogictest/test/index/orderby_nosort/10/slt_good_34.test with config fakedist-disk: 10052 tests, 0 failures
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/orderby_nosort/10/slt_good_31.test: 8183 statements/queries
--- total progress: 15102453 statements/queries
        --- FAIL: TestSqlLiteLogic/fakedist-disk/select4.test (219.51s)
            logic.go:1879: 
                 pq: this query requires additional disk space: in-mem temp storage: disk budget exceeded: 1048576 bytes requested, 104857600 currently allocated, 104857600 bytes in budget
            logic.go:2458: 
                /go/src/github.com/cockroachdb/sqllogictest/test/select4.test:38818: error while processing
            logic.go:2458: /go/src/github.com/cockroachdb/sqllogictest/test/select4.test:38818: too many errors encountered, skipping the rest of the input

```

<details><summary>More</summary><p>


```
make stressrace TESTS=TestSqlLiteLogic PKG=./pkg/sql/logictest TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestSqlLiteLogic.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(sql/logictest).TestSqlLiteLogic failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1922949&tab=buildLog) on [master@683f0d561bf9b73902edb27d681bca5333bdad60](https://github.com/cockroachdb/cockroach/commits/683f0d561bf9b73902edb27d681bca5333bdad60):

```
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/delete/1/slt_good_0.test: 9074 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/commute/100/slt_good_1.test: 2714 statements/queries
rewrote:
CREATE TABLE tab3 (pk INT8 PRIMARY KEY, col0 INT8, col1 FLOAT8, col2 STRING, col3 INT8, col4 FLOAT8, col5 STRING, FAMILY (col4, pk), FAMILY (col0), FAMILY (col5, col2, col3), FAMILY (col1));


--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/commute/100/slt_good_12.test: 9877 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/select4.test: 1123 statements/queries
rewrote:
CREATE TABLE tab4 (pk INT8 PRIMARY KEY, col0 INT8, col1 FLOAT8, col2 STRING, col3 INT8, col4 FLOAT8, col5 STRING, FAMILY (col1, col5, col0), FAMILY (col4, col3), FAMILY (pk), FAMILY (col2));


--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/orderby_nosort/100/slt_good_5.test: 1334 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/commute/100/slt_good_11.test: 7799 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/commute/100/slt_good_10.test: 2967 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/commute/100/slt_good_2.test: 9935 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/commute/100/slt_good_1.test: 2822 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/delete/1/slt_good_0.test: 9085 statements/queries
rewrote:
CREATE TABLE tab0 (pk INT8 PRIMARY KEY, col0 INT8, col1 FLOAT8, col2 STRING, col3 INT8, col4 FLOAT8, col5 STRING, FAMILY (col4), FAMILY (col2), FAMILY (pk), FAMILY (col5), FAMILY (col0, col1, col3));


rewrote:
CREATE TABLE tab1 (pk INT8 PRIMARY KEY, col0 INT8, col1 FLOAT8, col2 STRING, col3 INT8, col4 FLOAT8, col5 STRING, FAMILY (col1), FAMILY (col3, pk), FAMILY (col0, col2), FAMILY (col4), FAMILY (col5));


--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/commute/100/slt_good_12.test: 10012 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/select4.test: 1149 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/commute/100/slt_good_11.test: 7926 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/orderby_nosort/100/slt_good_5.test: 1475 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/commute/100/slt_good_10.test: 3053 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/commute/100/slt_good_2.test: 10045 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/commute/100/slt_good_1.test: 2921 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/delete/1/slt_good_0.test: 9113 statements/queries
rewrote:
CREATE TABLE tab2 (pk INT8 PRIMARY KEY, col0 INT8, col1 FLOAT8, col2 STRING, col3 INT8, col4 FLOAT8, col5 STRING, FAMILY (col5, col0, col4, col2, pk), FAMILY (col1, col3));


--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/commute/100/slt_good_12.test: 10110 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/select4.test: 1171 statements/queries
--- done: /go/src/github.com/cockroachdb/sqllogictest/test/index/commute/100/slt_good_12.test with config fakedist-vec-auto-disk: 10123 tests, 0 failures
--- total progress: 12199471 statements/queries
        --- FAIL: TestSqlLiteLogic/fakedist-vec-auto-disk/select4.test (217.49s)
            logic.go:1881: 
                 pq: this query requires additional disk space: in-mem temp storage: disk budget exceeded: 1048576 bytes requested, 104857600 currently allocated, 104857600 bytes in budget
            logic.go:2460: 
                /go/src/github.com/cockroachdb/sqllogictest/test/select4.test:38818: error while processing
            logic.go:2460: /go/src/github.com/cockroachdb/sqllogictest/test/select4.test:38818: too many errors encountered, skipping the rest of the input
=== RUN   TestSqlLiteLogic/fakedist-vec-auto-disk
    --- FAIL: TestSqlLiteLogic/fakedist-vec-auto-disk (0.00s)

```

<details><summary>More</summary><p>


```
make stressrace TESTS=TestSqlLiteLogic PKG=./pkg/sql/logictest TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestSqlLiteLogic.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(sql/logictest).TestSqlLiteLogic failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1930981&tab=buildLog) on [master@1d6db7e386e3ab21bca53f38637014f61153dc90](https://github.com/cockroachdb/cockroach/commits/1d6db7e386e3ab21bca53f38637014f61153dc90):

```

--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/commute/100/slt_good_8.test: 9167 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/commute/100/slt_good_9.test: 10041 statements/queries
rewrote:
CREATE TABLE tab1 (pk INT8 PRIMARY KEY, col0 INT8, col1 FLOAT8, col2 STRING, col3 INT8, col4 FLOAT8, col5 STRING, FAMILY (col1), FAMILY (col3, col2, pk, col0), FAMILY (col5, col4));


--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/in/100/slt_good_0.test: 5903 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/delete/10000/slt_good_0.test: 20331 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/delete/10/slt_good_0.test: 8532 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/delete/1/slt_good_0.test: 5734 statements/queries
rewrote:
CREATE TABLE tab0 (pk INT8 PRIMARY KEY, col0 INT8, col1 FLOAT8, col2 STRING, col3 INT8, col4 FLOAT8, col5 STRING, FAMILY (col3), FAMILY (pk, col5, col0), FAMILY (col2, col1, col4));


rewrote:
CREATE TABLE tab2 (pk INT8 PRIMARY KEY, col0 INT8, col1 FLOAT8, col2 STRING, col3 INT8, col4 FLOAT8, col5 STRING, FAMILY (col4), FAMILY (col3), FAMILY (col1), FAMILY (col5), FAMILY (col0, col2, pk));


rewrote:
CREATE TABLE tab1 (pk INT8 PRIMARY KEY, col0 INT8, col1 FLOAT8, col2 STRING, col3 INT8, col4 FLOAT8, col5 STRING, FAMILY (col3), FAMILY (col1, col4, col2, col0), FAMILY (col5, pk));


--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/commute/100/slt_good_9.test: 10106 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/commute/100/slt_good_8.test: 9233 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/select4.test: 2519 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/in/10/slt_good_3.test: 4254 statements/queries
--- done: /go/src/github.com/cockroachdb/sqllogictest/test/index/commute/100/slt_good_9.test with config fakedist-vec-auto-disk: 10123 tests, 0 failures
--- total progress: 11506797 statements/queries
        --- FAIL: TestSqlLiteLogic/fakedist-vec-auto-disk/select4.test (213.75s)
            logic.go:2126: 
                
                /go/src/github.com/cockroachdb/sqllogictest/test/select4.test:38805: SELECT e8, d7+959, b3, a5*693, e2, e6*447, b1+778, x4
                  FROM t4, t1, t6, t7, t5, t8, t2, t3
                 WHERE c5 in (489,250,927,855,697,130,856)
                   AND (180=e8 OR 333=e8 OR e8=38 OR e8=349)
                   AND e7=462
                   AND a3=d1
                   AND b2=b1
                   AND a1 in (358,451,479,382,330,281,953,732)
                   AND b4 in (434,700,94,860)
                   AND d6 in (489,911,147)
                expected success, but found
                (53100) this query requires additional disk space: in-mem temp storage: disk budget exceeded: 1048576 bytes requested, 104857600 currently allocated, 104857600 bytes in budget
                resource.go:57: in NewBudgetExceededError()
            logic.go:1881: 
                 pq: this query requires additional disk space: in-mem temp storage: disk budget exceeded: 1048576 bytes requested, 104857600 currently allocated, 104857600 bytes in budget
            logic.go:2459: 
                /go/src/github.com/cockroachdb/sqllogictest/test/select4.test:38818: error while processing
            logic.go:2459: /go/src/github.com/cockroachdb/sqllogictest/test/select4.test:38818: too many errors encountered, skipping the rest of the input

```

<details><summary>More</summary><p>


```
make stressrace TESTS=TestSqlLiteLogic PKG=./pkg/sql/logictest TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestSqlLiteLogic.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(sql/logictest).TestSqlLiteLogic failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1934007&tab=buildLog) on [master@e6b47088aba9c9978501b966a0e88aeb273d9990](https://github.com/cockroachdb/cockroach/commits/e6b47088aba9c9978501b966a0e88aeb273d9990):

```
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/orderby_nosort/10/slt_good_9.test: 9824 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/view/10/slt_good_1.test: 7380 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/select4.test: 1413 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/orderby_nosort/100/slt_good_0.test: 8876 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/orderby_nosort/10/slt_good_8.test: 6884 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/orderby_nosort/10/slt_good_7.test: 4933 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/view/10/slt_good_2.test: 7530 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/view/100/slt_good_3.test: 1133 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/orderby_nosort/10/slt_good_9.test: 9846 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/select4.test: 1443 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/view/10/slt_good_1.test: 7403 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/orderby_nosort/100/slt_good_0.test: 9024 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/orderby_nosort/10/slt_good_8.test: 6950 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/orderby_nosort/10/slt_good_7.test: 5037 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/view/100/slt_good_3.test: 1197 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/view/10/slt_good_2.test: 7562 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/orderby_nosort/10/slt_good_9.test: 9867 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/select4.test: 1472 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/view/10/slt_good_1.test: 7427 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/orderby_nosort/100/slt_good_0.test: 9225 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/orderby_nosort/10/slt_good_8.test: 7076 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/orderby_nosort/10/slt_good_7.test: 5120 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/view/100/slt_good_3.test: 1256 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/view/10/slt_good_2.test: 7586 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/orderby_nosort/10/slt_good_9.test: 10024 statements/queries
--- done: /go/src/github.com/cockroachdb/sqllogictest/test/index/orderby_nosort/10/slt_good_9.test with config fakedist-vec-auto-disk: 10055 tests, 0 failures
--- total progress: 11346947 statements/queries
        --- FAIL: TestSqlLiteLogic/fakedist-vec-auto-disk/select4.test (153.32s)
            logic.go:2126: 
                
                /go/src/github.com/cockroachdb/sqllogictest/test/select4.test:38805: SELECT e8, d7+959, b3, a5*693, e2, e6*447, b1+778, x4
                  FROM t4, t1, t6, t7, t5, t8, t2, t3
                 WHERE c5 in (489,250,927,855,697,130,856)
                   AND (180=e8 OR 333=e8 OR e8=38 OR e8=349)
                   AND e7=462
                   AND a3=d1
                   AND b2=b1
                   AND a1 in (358,451,479,382,330,281,953,732)
                   AND b4 in (434,700,94,860)
                   AND d6 in (489,911,147)
                expected success, but found
                (53100) this query requires additional disk space: in-mem temp storage: disk budget exceeded: 1048576 bytes requested, 104857600 currently allocated, 104857600 bytes in budget
                resource.go:57: in NewBudgetExceededError()
            logic.go:1881: 
                 pq: this query requires additional disk space: in-mem temp storage: disk budget exceeded: 1048576 bytes requested, 104857600 currently allocated, 104857600 bytes in budget
            logic.go:2459: 
                /go/src/github.com/cockroachdb/sqllogictest/test/select4.test:38818: error while processing
            logic.go:2459: /go/src/github.com/cockroachdb/sqllogictest/test/select4.test:38818: too many errors encountered, skipping the rest of the input
=== RUN   TestSqlLiteLogic/fakedist-vec-auto-disk
    --- FAIL: TestSqlLiteLogic/fakedist-vec-auto-disk (0.00s)

```

<details><summary>More</summary><p>


```
make stressrace TESTS=TestSqlLiteLogic PKG=./pkg/sql/logictest TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestSqlLiteLogic.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(sql/logictest).TestSqlLiteLogic failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1938748&tab=buildLog) on [master@3531318daf434670bb23ae68b8e8b8a4e4801879](https://github.com/cockroachdb/cockroach/commits/3531318daf434670bb23ae68b8e8b8a4e4801879):

```
CREATE TABLE t5 (a5 INT8, b5 INT8, c5 INT8, d5 INT8, e5 INT8, x5 VARCHAR(30), FAMILY (x5, c5, d5, e5, a5), FAMILY (b5));


rewrote:
CREATE TABLE t6 (a6 INT8, b6 INT8, c6 INT8, d6 INT8, e6 INT8, x6 VARCHAR(30), FAMILY (d6), FAMILY (x6, e6, b6, c6, a6));


rewrote:
CREATE TABLE t7 (a7 INT8, b7 INT8, c7 INT8, d7 INT8, e7 INT8, x7 VARCHAR(30), FAMILY (c7, x7), FAMILY (b7), FAMILY (e7), FAMILY (d7), FAMILY (a7));


rewrote:
CREATE TABLE t8 (a8 INT8, b8 INT8, c8 INT8, d8 INT8, e8 INT8, x8 VARCHAR(30), FAMILY (c8), FAMILY (a8, x8, e8, b8, d8));


rewrote:
CREATE TABLE t9 (a9 INT8, b9 INT8, c9 INT8, d9 INT8, e9 INT8, x9 VARCHAR(30), FAMILY (e9), FAMILY (x9), FAMILY (d9, a9), FAMILY (b9, c9));


--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/orderby_nosort/10/slt_good_25.test: 3152 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/orderby_nosort/10/slt_good_3.test: 9720 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/orderby_nosort/10/slt_good_28.test: 6034 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/orderby_nosort/10/slt_good_26.test: 3438 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/orderby_nosort/10/slt_good_27.test: 4920 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/orderby_nosort/10/slt_good_24.test: 823 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/view/10/slt_good_7.test: 320 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/select4.test: 602 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/orderby_nosort/10/slt_good_25.test: 3327 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/orderby_nosort/10/slt_good_3.test: 9829 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/orderby_nosort/10/slt_good_28.test: 6186 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/orderby_nosort/10/slt_good_26.test: 3646 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/orderby_nosort/10/slt_good_27.test: 5039 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/orderby_nosort/10/slt_good_24.test: 1015 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/view/10/slt_good_7.test: 402 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/select4.test: 1021 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/orderby_nosort/10/slt_good_25.test: 3544 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/orderby_nosort/10/slt_good_3.test: 9958 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/orderby_nosort/10/slt_good_28.test: 6369 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/orderby_nosort/10/slt_good_26.test: 3810 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/orderby_nosort/10/slt_good_27.test: 5260 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/orderby_nosort/10/slt_good_24.test: 1104 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/view/10/slt_good_7.test: 471 statements/queries
--- done: /go/src/github.com/cockroachdb/sqllogictest/test/index/orderby_nosort/10/slt_good_3.test with config fakedist-vec-auto-disk: 10051 tests, 0 failures
--- total progress: 11471609 statements/queries
        --- FAIL: TestSqlLiteLogic/fakedist-vec-auto-disk/select4.test (157.01s)
            logic.go:1881: 
                 pq: this query requires additional disk space: in-mem temp storage: disk budget exceeded: 1048576 bytes requested, 104857600 currently allocated, 104857600 bytes in budget
            logic.go:2459: 
                /go/src/github.com/cockroachdb/sqllogictest/test/select4.test:38818: error while processing
            logic.go:2459: /go/src/github.com/cockroachdb/sqllogictest/test/select4.test:38818: too many errors encountered, skipping the rest of the input

```

<details><summary>More</summary><p>


```
make stressrace TESTS=TestSqlLiteLogic PKG=./pkg/sql/logictest TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestSqlLiteLogic.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(sql/logictest).TestSqlLiteLogic failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1941958&tab=buildLog) on [master@752c32580ff452bb0039e5af0c00044b91bdcb12](https://github.com/cockroachdb/cockroach/commits/752c32580ff452bb0039e5af0c00044b91bdcb12):

```
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/commute/1000/slt_good_3.test: 10956 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/commute/10/slt_good_3.test: 7199 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/commute/10/slt_good_30.test: 7331 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/commute/10/slt_good_29.test: 4352 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/select4.test: 1049 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/delete/1/slt_good_0.test: 3751 statements/queries
rewrote:
CREATE TABLE tab0 (pk INT8 PRIMARY KEY, col0 INT8, col1 FLOAT8, col2 STRING, col3 INT8, col4 FLOAT8, col5 STRING, FAMILY (col3), FAMILY (col4, col2), FAMILY (col0, col1, col5), FAMILY (pk));


--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/commute/1000/slt_good_2.test: 9637 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/commute/10/slt_good_28.test: 2086 statements/queries
rewrote:
CREATE TABLE tab1 (pk INT8 PRIMARY KEY, col0 INT8, col1 FLOAT8, col2 STRING, col3 INT8, col4 FLOAT8, col5 STRING, FAMILY (col5, col4), FAMILY (pk), FAMILY (col3, col2), FAMILY (col0, col1));


--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/commute/1000/slt_good_3.test: 11009 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/commute/10/slt_good_3.test: 7262 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/commute/10/slt_good_30.test: 7425 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/commute/10/slt_good_29.test: 4474 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/select4.test: 1065 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/delete/1/slt_good_0.test: 3791 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/commute/1000/slt_good_2.test: 9704 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/commute/10/slt_good_28.test: 2153 statements/queries
--- done: /go/src/github.com/cockroachdb/sqllogictest/test/index/commute/1000/slt_good_3.test with config fakedist: 11025 tests, 0 failures
rewrote:
CREATE TABLE tab2 (pk INT8 PRIMARY KEY, col0 INT8, col1 FLOAT8, col2 STRING, col3 INT8, col4 FLOAT8, col5 STRING, FAMILY (col5), FAMILY (col2, col3), FAMILY (pk, col0, col4, col1));


--- total progress: 6099126 statements/queries
        --- FAIL: TestSqlLiteLogic/fakedist/select4.test (181.41s)
            logic.go:1881: 
                 
                /go/src/github.com/cockroachdb/sqllogictest/test/select4.test:38610: expected 735 results, but found 0
            logic.go:2459: 
                /go/src/github.com/cockroachdb/sqllogictest/test/select4.test:38622: error while processing
            logic.go:2459: /go/src/github.com/cockroachdb/sqllogictest/test/select4.test:38622: too many errors encountered, skipping the rest of the input
=== RUN   TestSqlLiteLogic/fakedist
    --- FAIL: TestSqlLiteLogic/fakedist (0.00s)
=== RUN   TestSqlLiteLogic/fakedist-vec-auto-disk/select4.test
=== PAUSE TestSqlLiteLogic/fakedist-vec-auto-disk/select4.test
=== CONT  TestSqlLiteLogic/fakedist-vec-auto-disk/select4.test
        --- FAIL: TestSqlLiteLogic/fakedist-vec-auto-disk/select4.test (174.94s)
            logic.go:1881: 
                 pq: this query requires additional disk space: in-mem temp storage: disk budget exceeded: 1048576 bytes requested, 104857600 currently allocated, 104857600 bytes in budget
            logic.go:2459: 
                /go/src/github.com/cockroachdb/sqllogictest/test/select4.test:38818: error while processing
            logic.go:2459: /go/src/github.com/cockroachdb/sqllogictest/test/select4.test:38818: too many errors encountered, skipping the rest of the input
=== RUN   TestSqlLiteLogic/fakedist-vec-auto-disk
    --- FAIL: TestSqlLiteLogic/fakedist-vec-auto-disk (0.00s)

```

<details><summary>More</summary><p>


```
make stressrace TESTS=TestSqlLiteLogic PKG=./pkg/sql/logictest TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestSqlLiteLogic.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(sql/logictest).TestSqlLiteLogic failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1947346&tab=buildLog) on [master@5156843cc23adecb6a70fabf19f51f46de1241ec](https://github.com/cockroachdb/cockroach/commits/5156843cc23adecb6a70fabf19f51f46de1241ec):

```
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/orderby_nosort/10/slt_good_1.test: 9906 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/view/10/slt_good_6.test: 3515 statements/queries
rewrote:
CREATE TABLE t1 (a1 INT8, b1 INT8, c1 INT8, d1 INT8, e1 INT8, x1 VARCHAR(30), FAMILY (a1), FAMILY (d1, b1, x1, e1, c1));


rewrote:
CREATE TABLE t2 (a2 INT8, b2 INT8, c2 INT8, d2 INT8, e2 INT8, x2 VARCHAR(30), FAMILY (x2), FAMILY (a2, c2), FAMILY (b2, e2, d2));


rewrote:
CREATE TABLE t3 (a3 INT8, b3 INT8, c3 INT8, d3 INT8, e3 INT8, x3 VARCHAR(30), FAMILY (d3, e3, x3), FAMILY (c3, a3), FAMILY (b3));


rewrote:
CREATE TABLE t4 (a4 INT8, b4 INT8, c4 INT8, d4 INT8, e4 INT8, x4 VARCHAR(30), FAMILY (x4, b4), FAMILY (a4, e4), FAMILY (d4, c4));


rewrote:
CREATE TABLE t5 (a5 INT8, b5 INT8, c5 INT8, d5 INT8, e5 INT8, x5 VARCHAR(30), FAMILY (d5, x5, a5, e5), FAMILY (b5), FAMILY (c5));


rewrote:
CREATE TABLE t6 (a6 INT8, b6 INT8, c6 INT8, d6 INT8, e6 INT8, x6 VARCHAR(30), FAMILY (d6), FAMILY (c6), FAMILY (e6, x6, b6), FAMILY (a6));


rewrote:
CREATE TABLE t7 (a7 INT8, b7 INT8, c7 INT8, d7 INT8, e7 INT8, x7 VARCHAR(30), FAMILY (x7), FAMILY (e7, d7), FAMILY (a7), FAMILY (c7), FAMILY (b7));


rewrote:
CREATE TABLE t8 (a8 INT8, b8 INT8, c8 INT8, d8 INT8, e8 INT8, x8 VARCHAR(30), FAMILY (d8), FAMILY (e8, a8, x8, c8), FAMILY (b8));


rewrote:
CREATE TABLE t9 (a9 INT8, b9 INT8, c9 INT8, d9 INT8, e9 INT8, x9 VARCHAR(30), FAMILY (b9, a9, e9, x9, d9), FAMILY (c9));


--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/commute/100/slt_good_8.test: 671 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/orderby/1000/slt_good_0.test: 6560 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/orderby/100/slt_good_3.test: 8309 statements/queries
--- done: /go/src/github.com/cockroachdb/sqllogictest/test/index/orderby_nosort/10/slt_good_1.test with config fakedist-vec-auto-disk: 10051 tests, 0 failures
        --- FAIL: TestSqlLiteLogic/fakedist-vec-auto-disk/select4.test (158.10s)
            logic.go:1952: 
                 pq: this query requires additional disk space: in-mem temp storage: disk budget exceeded: 1048576 bytes requested, 104857600 currently allocated, 104857600 bytes in budget
            logic.go:2530: 
                /go/src/github.com/cockroachdb/sqllogictest/test/select4.test:38818: error while processing
            logic.go:2530: /go/src/github.com/cockroachdb/sqllogictest/test/select4.test:38818: too many errors encountered, skipping the rest of the input
=== RUN   TestSqlLiteLogic/fakedist-vec-auto-disk
    --- FAIL: TestSqlLiteLogic/fakedist-vec-auto-disk (0.00s)

```

<details><summary>More</summary><p>


```
make stressrace TESTS=TestSqlLiteLogic PKG=./pkg/sql/logictest TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestSqlLiteLogic.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(sql/logictest).TestSqlLiteLogic failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1948978&tab=buildLog) on [master@2cbf620cd229a55622fe17fb15d20ada1dbcccd3](https://github.com/cockroachdb/cockroach/commits/2cbf620cd229a55622fe17fb15d20ada1dbcccd3):

```
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/orderby_nosort/100/slt_good_0.test: 9742 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/orderby_nosort/10/slt_good_4.test: 1564 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/view/100/slt_good_0.test: 737 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/orderby_nosort/10/slt_good_8.test: 9165 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/orderby_nosort/10/slt_good_5.test: 6381 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/orderby_nosort/10/slt_good_7.test: 9381 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/orderby_nosort/10/slt_good_6.test: 8204 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/orderby_nosort/100/slt_good_0.test: 9774 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/view/100/slt_good_0.test: 769 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/orderby_nosort/10/slt_good_4.test: 1568 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/select4.test: 1057 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/orderby_nosort/10/slt_good_8.test: 9235 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/orderby_nosort/10/slt_good_5.test: 6466 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/orderby_nosort/10/slt_good_7.test: 9443 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/orderby_nosort/10/slt_good_6.test: 8324 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/orderby_nosort/100/slt_good_0.test: 9851 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/view/100/slt_good_0.test: 823 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/orderby_nosort/10/slt_good_4.test: 1572 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/select4.test: 1062 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/orderby_nosort/10/slt_good_5.test: 6619 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/orderby_nosort/10/slt_good_8.test: 9397 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/orderby_nosort/10/slt_good_7.test: 9612 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/orderby_nosort/10/slt_good_6.test: 8427 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/orderby_nosort/100/slt_good_0.test: 9877 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/view/100/slt_good_0.test: 876 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/orderby_nosort/10/slt_good_4.test: 1577 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/orderby_nosort/10/slt_good_5.test: 6765 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/orderby_nosort/10/slt_good_8.test: 9534 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/select4.test: 1065 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/orderby_nosort/10/slt_good_7.test: 9814 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/orderby_nosort/10/slt_good_6.test: 8553 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/orderby_nosort/100/slt_good_0.test: 9901 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/view/100/slt_good_0.test: 927 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/orderby_nosort/10/slt_good_4.test: 1582 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/orderby_nosort/10/slt_good_8.test: 9678 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/orderby_nosort/10/slt_good_5.test: 6886 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/orderby_nosort/10/slt_good_7.test: 9978 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/orderby_nosort/10/slt_good_6.test: 8703 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/orderby_nosort/100/slt_good_0.test: 10073 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/select4.test: 1069 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/view/100/slt_good_0.test: 984 statements/queries
--- done: /go/src/github.com/cockroachdb/sqllogictest/test/index/orderby_nosort/10/slt_good_7.test with config fakedist: 10052 tests, 0 failures
--- total progress: 6134627 statements/queries
        --- FAIL: TestSqlLiteLogic/fakedist/select4.test (916.33s)
            logic.go:1952: 
                 
                /go/src/github.com/cockroachdb/sqllogictest/test/select4.test:45868: expected 384 results, but found 0
            logic.go:2530: 
                /go/src/github.com/cockroachdb/sqllogictest/test/select4.test:45879: error while processing
            logic.go:2530: /go/src/github.com/cockroachdb/sqllogictest/test/select4.test:45879: too many errors encountered, skipping the rest of the input

```

<details><summary>More</summary><p>


```
make stressrace TESTS=TestSqlLiteLogic PKG=./pkg/sql/logictest TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestSqlLiteLogic.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(sql/logictest).TestSqlLiteLogic failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1950185&tab=buildLog) on [master@7ab7fb86f0634df1f5a0b04460e1f3a1d6bead1f](https://github.com/cockroachdb/cockroach/commits/7ab7fb86f0634df1f5a0b04460e1f3a1d6bead1f):

```
=== RUN   TestSqlLiteLogic/fakedist-disk/select4.test
=== PAUSE TestSqlLiteLogic/fakedist-disk/select4.test
=== CONT  TestSqlLiteLogic/fakedist-disk/select4.test
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/commute/10/slt_good_20.test: 5959 statements/queries
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/commute/10/slt_good_19.test: 1564 statements/queries
--- done: /go/src/github.com/cockroachdb/sqllogictest/test/index/commute/10/slt_good_22.test with config fakedist-disk: 10033 tests, 0 failures
rewrote:
CREATE TABLE t1 (a1 INT8, b1 INT8, c1 INT8, d1 INT8, e1 INT8, x1 VARCHAR(30), FAMILY (b1, c1), FAMILY (d1, a1, x1), FAMILY (e1));


--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/delete/1/slt_good_0.test: 8471 statements/queries
rewrote:
CREATE TABLE t2 (a2 INT8, b2 INT8, c2 INT8, d2 INT8, e2 INT8, x2 VARCHAR(30), FAMILY (e2), FAMILY (d2), FAMILY (b2, a2), FAMILY (x2), FAMILY (c2));


rewrote:
CREATE TABLE t3 (a3 INT8, b3 INT8, c3 INT8, d3 INT8, e3 INT8, x3 VARCHAR(30), FAMILY (e3), FAMILY (a3), FAMILY (x3, b3), FAMILY (d3, c3));


rewrote:
CREATE TABLE t4 (a4 INT8, b4 INT8, c4 INT8, d4 INT8, e4 INT8, x4 VARCHAR(30), FAMILY (c4, e4, d4), FAMILY (a4), FAMILY (b4), FAMILY (x4));


rewrote:
CREATE TABLE t5 (a5 INT8, b5 INT8, c5 INT8, d5 INT8, e5 INT8, x5 VARCHAR(30), FAMILY (d5), FAMILY (a5), FAMILY (x5, b5), FAMILY (c5, e5));


        --- FAIL: TestSqlLiteLogic/fakedist-disk/select4.test (831.32s)
            logic.go:2197: 
                
                /go/src/github.com/cockroachdb/sqllogictest/test/select4.test:38805: SELECT e8, d7+959, b3, a5*693, e2, e6*447, b1+778, x4
                  FROM t4, t1, t6, t7, t5, t8, t2, t3
                 WHERE c5 in (489,250,927,855,697,130,856)
                   AND (180=e8 OR 333=e8 OR e8=38 OR e8=349)
                   AND e7=462
                   AND a3=d1
                   AND b2=b1
                   AND a1 in (358,451,479,382,330,281,953,732)
                   AND b4 in (434,700,94,860)
                   AND d6 in (489,911,147)
                expected success, but found
                (53100) in-mem temp storage: disk budget exceeded: 1048576 bytes requested, 104857600 currently allocated, 104857600 bytes in budget
                resource.go:57: in NewBudgetExceededError()
            logic.go:1952: 
                 pq: in-mem temp storage: disk budget exceeded: 1048576 bytes requested, 104857600 currently allocated, 104857600 bytes in budget
            logic.go:2530: 
                /go/src/github.com/cockroachdb/sqllogictest/test/select4.test:38818: error while processing
            logic.go:2530: /go/src/github.com/cockroachdb/sqllogictest/test/select4.test:38818: too many errors encountered, skipping the rest of the input
=== RUN   TestSqlLiteLogic/fakedist-disk
    --- FAIL: TestSqlLiteLogic/fakedist-disk (0.00s)

```

<details><summary>More</summary><p>


```
make stressrace TESTS=TestSqlLiteLogic PKG=./pkg/sql/logictest TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestSqlLiteLogic.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

--- a/pkg/base/test_server_args.go
+++ b/pkg/base/test_server_args.go
@@ -150,7 +150,14 @@ var (
 // DefaultTestStoreSpec that is in-memory.
 // It has a maximum size of 100MiB.
 func DefaultTestTempStorageConfig(st *cluster.Settings) TempStorageConfig {
-	var maxSizeBytes int64 = DefaultInMemTempStorageMaxSizeBytes
+	return DefaultTestTempStorageConfigWithSize(st, DefaultInMemTempStorageMaxSizeBytes)
+}
+
+// DefaultTestTempStorageConfigWithSize is the associated temp storage for
+// DefaultTestStoreSpec that is in-memory with the customized maximum size.
+func DefaultTestTempStorageConfigWithSize(
+	st *cluster.Settings, maxSizeBytes int64,
+) TempStorageConfig {
 	monitor := mon.MakeMonitor(
 		"in-mem temp storage",
 		mon.DiskResource,

--- a/pkg/ccl/logictestccl/logic_test.go
+++ b/pkg/ccl/logictestccl/logic_test.go
@@ -18,5 +18,5 @@ import (
 
 func TestCCLLogic(t *testing.T) {
 	defer leaktest.AfterTest(t)()
-	logictest.RunLogicTest(t, "testdata/logic_test/[^.]*")
+	logictest.RunLogicTest(t, logictest.TestServerArgs{}, "testdata/logic_test/[^.]*")
 }

--- a/pkg/sql/logictest/logic.go
+++ b/pkg/sql/logictest/logic.go
@@ -1162,7 +1162,7 @@ func (t *logicTest) setUser(user string) func() {
 	return cleanupFunc
 }
 
-func (t *logicTest) setup(cfg testClusterConfig) {
+func (t *logicTest) setup(cfg testClusterConfig, serverArgs TestServerArgs) {
 	t.cfg = cfg
 	// TODO(pmattis): Add a flag to make it easy to run the tests against a local
 	// MySQL or Postgres instance.
@@ -1170,11 +1170,18 @@ func (t *logicTest) setup(cfg testClusterConfig) {
 	// it installs detects a transaction that doesn't have
 	// modifiedSystemConfigSpan set even though it should, for
 	// "testdata/rename_table". Figure out what's up with that.
+	var tempStorageConfig base.TempStorageConfig
+	if serverArgs.tempStorageDiskLimit == 0 {
+		tempStorageConfig = base.DefaultTestTempStorageConfig(cluster.MakeTestingClusterSettings())
+	} else {
+		tempStorageConfig = base.DefaultTestTempStorageConfigWithSize(cluster.MakeTestingClusterSettings(), serverArgs.tempStorageDiskLimit)
+	}
 	params := base.TestClusterArgs{
 		ServerArgs: base.TestServerArgs{
 			// Specify a fixed memory limit (some test cases verify OOM conditions; we
 			// don't want those to take long on large machines).
 			SQLMemoryPoolSize: 192 * 1024 * 1024,
+			TempStorageConfig: tempStorageConfig,
 			Knobs: base.TestingKnobs{
 				Store: &kvserver.StoreTestingKnobs{
 					// The consistency queue makes a lot of noisy logs during logic tests.
@@ -2535,9 +2542,18 @@ var skipLogicTests = envutil.EnvOrDefaultBool("COCKROACH_LOGIC_TESTS_SKIP", fals
 var logicTestsConfigExclude = envutil.EnvOrDefaultString("COCKROACH_LOGIC_TESTS_SKIP_CONFIG", "")
 var logicTestsConfigFilter = envutil.EnvOrDefaultString("COCKROACH_LOGIC_TESTS_CONFIG", "")
 
+// TestServerArgs contains the parameters that callers of RunLogicTest might
+// want to specify for the test clusters to be created with.
+type TestServerArgs struct {
+	// tempStorageDiskLimit determines the limit for the temp storage (that is
+	// actually in-memory). If it is unset, then the default limit of 100MB
+	// will be used.
+	tempStorageDiskLimit int64
+}
+
 // RunLogicTest is the main entry point for the logic test. The globs parameter
 // specifies the default sets of files to run.
-func RunLogicTest(t *testing.T, globs ...string) {
+func RunLogicTest(t *testing.T, serverArgs TestServerArgs, globs ...string) {
 	// Note: there is special code in teamcity-trigger/main.go to run this package
 	// with less concurrency in the nightly stress runs. If you see problems
 	// please make adjustments there.
@@ -2665,7 +2681,7 @@ func RunLogicTest(t *testing.T, globs ...string) {
 					if *printErrorSummary {
 						defer lt.printErrorSummary()
 					}
-					lt.setup(cfg)
+					lt.setup(cfg, serverArgs)
 					lt.runFile(path, cfg)
 
 					progress.Lock()

--- a/pkg/sql/logictest/logic_test.go
+++ b/pkg/sql/logictest/logic_test.go
@@ -29,7 +29,7 @@ import (
 // See the comments in logic.go for more details.
 func TestLogic(t *testing.T) {
 	defer leaktest.AfterTest(t)()
-	RunLogicTest(t, "testdata/logic_test/[^.]*")
+	RunLogicTest(t, TestServerArgs{}, "testdata/logic_test/[^.]*")
 }
 
 // TestSqlLiteLogic runs the supported SqlLite logic tests. See the comments
@@ -105,5 +105,10 @@ func runSQLLiteLogicTest(t *testing.T, globs ...string) {
 		prefixedGlobs[i] = logicTestPath + glob
 	}
 
-	RunLogicTest(t, prefixedGlobs...)
+	// SQLLite logic tests can be very disk (with '-disk' configs) intensive,
+	// so we give them larger temp storage limit than other logic tests get.
+	serverArgs := TestServerArgs{
+		tempStorageDiskLimit: 512 << 20, // 512 MiB
+	}
+	RunLogicTest(t, serverArgs, prefixedGlobs...)
 }

--- a/pkg/sql/opt/exec/execbuilder/builder_test.go
+++ b/pkg/sql/opt/exec/execbuilder/builder_test.go
@@ -27,5 +27,5 @@ import (
 func TestExecBuild(t *testing.T) {
 	defer leaktest.AfterTest(t)()
 	defer sql.TestingOverrideExplainEnvVersion("CockroachDB execbuilder test version")()
-	logictest.RunLogicTest(t, "testdata/[^.]*")
+	logictest.RunLogicTest(t, logictest.TestServerArgs{}, "testdata/[^.]*")
 }

